server/sonar-server/src/main/java/org/sonar/server/measure/ws/SearchAction.java

```diff
@@ -124,43 +123,38 @@ private SearchRequest createRequest() {
       return request;
     }
 
-    private List<MetricDto> searchMetrics() {
-      List<MetricDto> dbMetrics = dbClient.metricDao().selectByKeys(dbSession, request.getMetricKeys());
-      List<String> metricKeys = dbMetrics.stream().map(MetricDto::getKey).collect(toList());
-      checkRequest(request.getMetricKeys().size() == dbMetrics.size(), "The following metrics are not found: %s",
-        String.join(", ", difference(request.getMetricKeys(), metricKeys)));
-      return dbMetrics;
-    }
-
     private List<ComponentDto> searchProjects() {
       return getAuthorizedProjects(searchByProjectKeys(dbSession, request.getProjectKeys()));
     }
 
-    private List<ComponentDto> getAuthorizedProjects(List<ComponentDto> projectDtos) {
+    private List<ComponentDto> searchByProjectKeys(DbSession dbSession, List<String> projectKeys) {
```

Comment:
do you mind changing the check on the max number of elements, so it's really limited to 100 elements and not 99? This is the bug pointed by Stas